app/models/game.rb

```diff
@@ -21,4 +22,18 @@ def populate_board!
       count += 1
     end
   end
+
+  def white_king_check?
+    pieces.where(color: 'black').each do |piece|
+      return true if piece.valid_move?(@white_king.horizontal_position, @white_king.vertical_position)
+      false
+    end
+  end
+
+  def black_king_check?
+    pieces.where(color: 'white').each do |piece|
+      return true if piece.valid_move?(@black_king.horizontal_position, @black_king.vertical_position)
+      false
+    end
+  end
```

Comment:
We can combine white_king_check? and black_king_check? into one method (say king_check?) that takes a parameter for the color and then sets a local var king to either @white_king or @black_king. That way we can avoid the code duplication.